daemon/export.go

```diff
@@ -71,11 +71,18 @@ func (daemon *Daemon) containerExport(ctx context.Context, ctr *container.Contai
 		return err
 	}
 
-	if err := ctx.Err(); err != nil {
-		return err
-	}
+	ctx, cancel := context.WithCancel(ctx)
```

Comment:
Wondering if  this is  doing something similar to `ioutils.NewCancelReadCloser` ?

Reply:
Similar, but it's less heavy as we don't need the extra `io.Pipe()` here

Reply:
ah! Yes, guess that makes sense 👍

Reply:
Actually, there are 2 usages of `ioutils.NewCancelReadCloser` and I _think_ they could also be simplified to not use the `io.Pipe`.. But would need to double check that.